src/system/Form/InputWithCopyButton.js

```diff
@@ -0,0 +1,107 @@
+/** @jsxImportSource theme-ui */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { MdContentCopy } from 'react-icons/md';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Validation, Label } from '..';
+import { Input as ThemeInput } from 'theme-ui';
+import { baseControlStyle } from './Input.styles';
+
+const inputStyles = {
+	unset: 'all',
+	...baseControlStyle,
+	lineHeight: 'inherit',
+	minHeight: '36px',
+	px: 3,
+	py: 2,
+	fontSize: 2,
+	mb: 2,
+	variant: 'inputs.default',
+};
+
+const InputWithCopyButton = React.forwardRef(
+	(
+		{ variant, label, forLabel, hasError = false, required, sx = {}, errorMessage, ...props },
+		ref
+	) => {
+		if ( ! ref ) {
+			ref = React.createRef();
+		}
+
+		const handleCopy = e => {
+			e.preventDefault();
+			if ( ! ref.current ) {
+				return;
+			}
+			const originalTypeStatus = ref.current.getAttribute( 'type' ) || '';
+			if ( ref.current.getAttribute( 'value' ) === '' ) {
+				ref.current.setAttribute( 'value', '    ' );
+			}
+			ref.current.setAttribute( 'type', 'text' );
+			ref.current.select();
+			document.execCommand( 'copy' ); // eslint-disable-line no-undef
+			ref.current?.setAttribute( 'type', originalTypeStatus );
+		};
+		return (
+			<React.Fragment>
+				{ label && (
+					<Label required={ required } htmlFor={ forLabel }>
+						{ label }
+					</Label>
+				) }
+				<div sx={ { display: 'flex' } }>
+					<ThemeInput
+						ref={ ref }
+						id={ forLabel }
+						required={ required }
+						aria-required={ required }
+						aria-describedby={ hasError ? `describe-${ forLabel }-validation` : undefined }
+						sx={ {
+							...inputStyles,
+							...sx,
+							...( hasError ? { borderColor: 'input.border.error' } : {} ),
+						} }
+						{ ...props }
+					/>
+					<div sx={ { ml: 2 } }>
+						<Button
+							sx={ { height: '40px' } }
+							aria-label={ `Copy ${ label }` }
+							onClick={ handleCopy }
+							variant="ghost"
+						>
+							<MdContentCopy sx={ { mr: 2 } } />
+							Copy
```

Comment:
@kat3samsin I tested it live and I missed having some kind of feedback that the data was actually copied somewhere.
Something like "Copied !", which usually replaces the `Copy` message.

I'm not sure, though, how we should approach this from an accessibility perspective. 
Should we use`aria-atomic` so that it's announced when the content changes? 
What alternatives do you see here?

Reply:
@kat3samsin good idea! I like it ;)

Reply:
I added the notification callback.
<img width="980" alt="image" src="https://github.com/Automattic/vip-design-system/assets/2224344/7f7352ca-bd67-4e8d-ba8c-d7b1a0467a6d">